File: python/fundamentals/day-04/Problem5.py
import math

num = int(input("Input:\n"))

is_prime = True

if num <= 1:
    is_prime = False
else:
    for i in range(2, int(math.isqrt(num)) + 1):
        if num % i == 0:
            is_prime = False
            break

if is_prime:
    print("Prime Number")
else:
    print("Not a Prime Number")

File: java/fundamentals/day04/Problem5.java
package fundamentals.day04;
import java.util.Scanner;
public class Problem5 {
	

	    public static void main(String[] args) {
	        Scanner scanner = new Scanner(System.in);
	        System.out.println("Input:");
	        int num = scanner.nextInt();

	        boolean isPrime = true;

	        if (num <= 1) {
	            isPrime = false;
	        } else {
	            for (int i = 2; i <= Math.sqrt(num); i++) {
	                if (num % i == 0) {
	                    isPrime = false;
	                    break;
	                }
	            }
	        }

	        if (isPrime) {
	            System.out.println("Prime Number");
	        } else {
	            System.out.println("Not a Prime Number");
	        }

	        scanner.close();
	    }
	}

